### PR TITLE
API-6746: First version of domain library with items taken directly f…

### PR DESCRIPTION
**Pull request by mattclark-zerogravit**
…rom api-platform-application-events repo

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+val appName = "api-platform-application-domain"
+lazy val scala212 = "2.12.15"
+lazy val scala213 = "2.13.8"
+lazy val supportedScalaVersions = List(scala212, scala213)
+
+ThisBuild / scalafixDependencies += "com.github.liancheng" %% "organize-imports" % "0.6.0"
+ 
+inThisBuild(
+  List(
+    scalaVersion := scala213,
```
Comment (peteslater-ee) [inline build.sbt:17]: You don't need this

File: src/main/scala/uk/gov/hmrc/apiplatform/modules/applications/domain/models/ApplicationId.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.apiplatform.modules.applications.domain.models
+
+import java.util.UUID
+
+import play.api.libs.json.Json
+
+/** This file should be in some Application based library
```
Comment (peteslater-ee) [inline src/main/scala/uk/gov/hmrc/apiplatform/modules/applications/domain/models/ApplicationId.scala:23]: It is now right?

Comment (mattclark-zerogravit) [reply src/main/scala/uk/gov/hmrc/apiplatform/modules/applications/domain/models/ApplicationId.scala:23]: oops copy and paste :)

Comment (mattclark-zerogravit) [reply src/main/scala/uk/gov/hmrc/apiplatform/modules/applications/domain/models/ApplicationId.scala]: Have addressed

**Review by peteslater-ee — APPROVED** [review]

**Pull request merged**